Apps/WebClient/src/ClientApp/src/components/timeline/calendarTimeline.vue

```diff
@@ -1,54 +1,39 @@
 <script lang="ts">
 import Vue from "vue";
 import { Component, Prop, Watch } from "vue-property-decorator";
+import { Getter } from "vuex-class";
 
 import CalendarComponent from "@/components/calendar/calendar.vue";
 import TimelineEntry, { DateGroup } from "@/models/timelineEntry";
-import TimelineFilter from "@/models/timelineFilter";
+
 @Component({
     components: {
         CalendarComponent,
     },
 })
 export default class CalendarTimelineComponent extends Vue {
     @Prop() private timelineEntries!: TimelineEntry[];
-    @Prop() private isVisible!: boolean;
-    @Prop() private totalEntries!: number;
-    @Prop() private filter!: TimelineFilter;
 
-    private filteredTimelineEntries: TimelineEntry[] = [];
-    private dateGroups: DateGroup[] = [];
-    private hasFilter = false;
+    @Getter("hasActiveFilter", { namespace: "timeline" })
+    hasActiveFilter!: boolean;
 
-    @Watch("filter", { deep: true })
-    private applyTimelineFilter() {
-        this.hasFilter = this.filter.hasActiveFilter();
-        this.filteredTimelineEntries = this.timelineEntries.filter((entry) =>
-            entry.filterApplies(this.filter)
-        );
+    private dateGroups: DateGroup[] = [];
 
-        this.dateGroups = DateGroup.createGroups(this.filteredTimelineEntries);
-        this.dateGroups = DateGroup.sortGroup(this.dateGroups, false);
+    private get timelineIsEmpty(): boolean {
+        return this.timelineEntries.length === 0;
     }
 
     @Watch("timelineEntries")
-    private refreshEntries() {
-        this.applyTimelineFilter();
-    }
-
-    private get timelineIsEmpty(): boolean {
-        return this.filteredTimelineEntries.length == 0;
+    private refreshTimelineGroups() {
+        this.dateGroups = DateGroup.createGroups(this.timelineEntries);
+        this.dateGroups = DateGroup.sortGroups(this.dateGroups, false);
```

Comment:
I think this could cause double rendering since we change the dateGroups twice, I think this is what was happening with the bugs on reports, lets create a variable and assigned it to the property after is being changed.

Reply:
just realized this was also there before your changes, but might be a good improvement?!

Reply:
good suggestion, I've updated them